xarray/plot/plot.py

```diff
@@ -633,7 +633,9 @@ def newplotfunc(
 
         # Decide on a default for the colorbar before facetgrids
         if add_colorbar is None:
-            add_colorbar = plotfunc.__name__ != "contour"
+            add_colorbar = plotfunc.__name__ != "contour" and not (
+                plotfunc.__name__ == "surface" and cmap is None
+            )
```

Comment:
```suggestion
            add_colorbar = True
            if plotfunc.__name__ != "contour" or (plotfunc.__name__ == "surface" and cmap is None):
                add_colorbar  = False
```

The other solution is more elegant but harder to understand (for me) - did I get the logic right?

Reply:
I think it should be
```suggestion
            add_colorbar = True
            if plotfunc.__name__ == "contour" or (
                plotfunc.__name__ == "surface" and cmap is None
            ):
                add_colorbar  = False
```